### DISCUSSION TITLE
How to prevent access to GraphQL SDL endpoint?

### DISCUSSION THREAD

**Discussion by joelamltc** [directus/directus#16467] [Archive]
Currently, getting the GraphQL SDL from the endpoint is not guarded by authentication. However, the documentation is saying that we can only get it when authenticated. https://docs.directus.io/reference/system/server.html#get-graphql-sdl.

Any thoughts? How can I completely block access to the /server/specs/graphql endpoint?

**Comment by daawaan4x** [comment]
AFAIK you can't completely lock the endpoint BUT the SDL returned is only specific to the current auth permissions present with the user requesting the resource (e.g. if you have collections A & B and only collection A is not accessible to the user requesting, then only collection B will be present in the SDL returned)

**Reply by rijkvanzanten** [reply]
That's correct 👍🏻

**Comment by abdonrd** [comment]
Even though access is restricted based on user role...
Is there no way to block it completely?

**Reply by rijkvanzanten** [reply]
Not as of right now. The specs endpoints return the specs for the currently authenticated user, but there are no global settings to enable/disable it completely

**Reply by abdonrd** [reply]
Can we turn this into a feature request? Or should we create a new discussion?

**Reply by rijkvanzanten** [reply]
Lets create a new feature request discussion for that 👍🏻 This one doesn't have the RFC format we'd like to use 🙂
